# **BIG MART SALES** 


**Contexto de Negocio**:
* BigMartSales es un dataset de ventas proveniente de una empresa del rubro retail, en el cual figuran la venta de los diferentes productos presentes en las 10 diferentes tiendas. 
* El objetivo del presente estudio es poder predecir las ventas de cada producto en un punto de venta en particular.

**Las características que se disponibilizan son las siguientes:**
- **Item_Identifier** --- (ID de producto unico.)
- **Item_Weight** --- (Peso del producto)
- **Item_Fat_Content** --- (EL producto es bajo en grasa o no?)
- **Item_Visibility** --- (El % del área de visualización total de todos los productos en una tienda asignada a un producto en particular)
- **Item_Type** --- (La categoria a la que pertenece el producto)
- **Item_MRP** --- (Precio minorista maximo -precio de lista- del producto)
- **Outlet_Identifier** --- (ID de tienda unica)
- **Outlet_Establishment_Year** --- (EL año en que se establecio la tienda)
- **Outlet_Size** --- (El tamaño de la tienda en terminos de superficie cubierta)
- **Outlet_Location_Type** --- (El tipo  de ciudad en la que se encuentra la tienda)
- **Outlet_Type** --- (Si la tienda es solo una tienda de comestibles o algun tipo de supermercado)

In [82]:
#Importando librerias necesarias para el Analisis Exploratorio
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns 
import warnings 
warnings.filterwarnings('ignore') 

In [83]:
#Extrayendo la data y  mostrando la cabecera
big_mart = "E:\FONDO TALENTO DIGITAL\Portafolio_GITHUB\Big Mart Sales Prediction\Big-Mart-Sales-Prediction\BigMartSales.csv"
df = pd.read_csv(big_mart)
df.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

In [84]:
#Mostrando las ultimas 3 filas del Dataframe
df.tail(3)

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
8520           NCJ29        10.60          Low Fat         0.035186   
8521           FDN46         7.21          Regular         0.145221   
8522           DRG01        14.80          Low Fat         0.044878   

               Item_Type  Item_MRP Outlet_Identifier  \
8520  Health and Hygiene   85.1224            OUT035   
8521         Snack Foods  103.1332            OUT018   
8522         Soft Drinks   75.4670            OUT046   

      Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
8520                       2004       Small               Tier 2   
8521                       2009      Medium               Tier 3   
8522                       1997       Small               Tier 1   

            Outlet_Type  Item_Outlet_Sales  
8520  Supermarket Type1          1193.1136  
8521  Supermarket Type2          1845.5976  
8522  Supermarket Type1           765.6700

In [85]:
#Verificando la cantidad de filas y características presentes en el dataframe
df.shape
print(f"El Dataframe contiene: '{df.shape[0]}' filas")
print(f"El Dataframe contiene: '{df.shape[1]}' características")

El Dataframe contiene: '8523' filas
El Dataframe contiene: '12' características


In [86]:
#Verificando cantidad de filas duplicadas
df.duplicated().sum()

0

In [87]:
#Información general del Dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


In [88]:
#Cantidad de valores unicos por columna
df.nunique()

Item_Identifier              1559
Item_Weight                   415
Item_Fat_Content                5
Item_Visibility              7880
Item_Type                      16
Item_MRP                     5938
Outlet_Identifier              10
Outlet_Establishment_Year       9
Outlet_Size                     3
Outlet_Location_Type            3
Outlet_Type                     4
Item_Outlet_Sales            3493
dtype: int64

In [89]:
#verificando nulos por características
df.isnull().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

In [90]:
# Porcentaje de valores nulos por columna
cant_valores_nulos = df.isnull().sum().sort_values(ascending=False)
porcentaje_nulos = (cant_valores_nulos/df.shape[0])*100
porcentaje_nulos.round(2)

Outlet_Size                  28.28
Item_Weight                  17.17
Item_Identifier               0.00
Item_Fat_Content              0.00
Item_Visibility               0.00
Item_Type                     0.00
Item_MRP                      0.00
Outlet_Identifier             0.00
Outlet_Establishment_Year     0.00
Outlet_Location_Type          0.00
Outlet_Type                   0.00
Item_Outlet_Sales             0.00
dtype: float64

- *Observamos que el porcentaje de valores nulos en la columna Outlet_Size es del 28.28% del total de la columna.*
- *El porcentaje de la columna de Item_Weight de valores nulos tambien es inferior ya que esta representado por el 17.17% del total.*
- *En base a estos valores inferiores se decide mantener las columnas para el Analisis Exploratorio de los datos, para esto se buscara realizar la limpieza e imputación adecuados.* 

In [91]:
#Valor promedio de la columna Item_Weight
df['Item_Weight'].value_counts().mean()

17.012048192771083

In [92]:
#Cantidad de valores presentes en la columna Outlet_Size
df['Outlet_Size'].value_counts()

Outlet_Size
Medium    2793
Small     2388
High       932
Name: count, dtype: int64

## **Analisis de Valores nulos**
+ Para valores **Categoricos**, reemplazaremos los nulos por la moda
+ Para valores **Numericos**, reemplazaremos los nulos por el promedio

### **Analisis y reemplazo de valores nulos para la columna numerica Item_weight**

In [93]:
#Reemplazo de valores nulos en la columna numerica Item_Weight
df['Item_Weight'].fillna(df['Item_Weight'].mean(),inplace=True)
df['Item_Weight'].isnull().sum()

0

### **Analisis y reemplazo de valores nulos para la columna categorica Outlet_Size**

In [94]:
#Conteo de valores presentes en la columna Outlet_Size
df['Outlet_Size'].value_counts()

Outlet_Size
Medium    2793
Small     2388
High       932
Name: count, dtype: int64

In [97]:
#Valor que mas se repite dentro de la columna Outlet_Size
df['Outlet_Size'].mode()

0    Medium
Name: Outlet_Size, dtype: object

In [99]:
#Reemplazo de valores nulos en la columna Outlet_Size por la moda = Medium
df['Outlet_Size'].fillna("Medium",inplace=True)
df['Outlet_Size'].isnull().sum()

0

### **Dataframe libre de valores nulos**

In [100]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                8523 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                8523 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


*Como se observa, el Dataframe se encuentra libre de valores nulos y como siguientes pasos se realizara lo siguiente*:
+ Analisis de las variables categoricas
+ Analisis de las variables numericas